tests/e2e/config/jest.setup.js

```diff
@@ -37,41 +39,29 @@ async function trashExistingPosts() {
 }
 
 /**
- * Navigates to the product listing screen and bulk-trashes any product which exist.
+ * Use api package to delete products.
  *
  * @return {Promise} Promise resolving once products have been trashed.
  */
-async function trashExistingProducts() {
-	await merchant.login();
-	// Visit `/wp-admin/edit.php?post_type=product` so we can see a list of products and delete them.
-	await visitAdminPage( 'edit.php', 'post_type=product' );
+async function deleteAllProducts() {
+	const repository = SimpleProduct.restRepository( factories.api.withDefaultPermalinks );
+	let products;
 
-	// If this selector doesn't exist there are no products for us to delete.
-	const bulkSelector = await page.$( '#bulk-action-selector-top' );
-	if ( ! bulkSelector ) {
-		return;
+	products = await repository.list();
+	while ( products.length > 0 ) {
```

Comment:
I don't know if this will add much more optimization over this loop, but I wonder if we could consider adding support for the batch update endpoint which allows us to pass in an array of product IDs in the future.

Reply:
So maybe something like `repository.batchUpdate( action: string, ids: array )`. Action could be any of `create`, `update` and `delete`, or we could just do a single action method liek `repository.batchDelete( ids: array )`.

Reply:
The API does support [batch processing](https://woocommerce.github.io/woocommerce-rest-api-docs/#batch-update-products) so this is something we should do in the 0.3 version of the package.